From 7915e60c510770ce8c6a36ee628547330e733de4 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 28 Nov 2018 16:27:39 +0800
Subject: [PATCH 5226/5242] MLK-16302-1 drm/imx: dpu: plane: Free dpu_plane
 when bailout from dpu_plane_init

commit  aed3e47d016be89024df397fbea6f43ce3384ca3 from
https://source.codeaurora.org/external/imx/linux-imx.git

The dpu_plane is allocated with kzalloc() in dpu_plane_init().
In case, dpu_plane_init() fails after that allocation, the bailout path
should free the dpu_plane, otherwise, there will be memory leakage.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
(cherry picked from commit 99cb80f855aafbe524519db635c3d03bd4cc8d85)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-plane.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-plane.c b/drivers/gpu/drm/imx/dpu/dpu-plane.c
index 0a62b00..c14f9db 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-plane.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-plane.c
@@ -988,8 +988,10 @@ struct dpu_plane *dpu_plane_init(struct drm_device *drm,
 		ret = -EINVAL;
 	}
 
-	if (ret)
+	if (ret) {
+		kfree(dpu_plane);
 		return ERR_PTR(ret);
+	}
 
 	return dpu_plane;
 }
-- 
1.7.9.5

